File: app/src/main/java/com/genonbeta/TrebleShot/migration/db/object/WritablePathObjectV12.java
/*
 * Copyright (C) 2020 Veli Tasalı
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

package com.genonbeta.TrebleShot.migration.db.object;

import android.content.ContentValues;
import android.database.sqlite.SQLiteDatabase;
import android.net.Uri;
import com.genonbeta.android.database.DatabaseObject;
import com.genonbeta.android.database.KuickDb;
import com.genonbeta.android.database.Progress;
import com.genonbeta.android.database.SQLQuery;

import static com.genonbeta.TrebleShot.migration.db.Migration.v12.*;

/**
 * created by: Veli
 * date: 16.02.2018 12:56
 */

public class WritablePathObjectV12 implements DatabaseObject<Object>
{
    public String title;
    public Uri path;

    public WritablePathObjectV12()
    {
    }

    public WritablePathObjectV12(Uri path)
    {
        this.path = path;
    }

    public WritablePathObjectV12(String title, Uri path)
    {
        this(path);
        this.title = title;
    }

    @Override
    public SQLQuery.Select getWhere()
    {
        return new SQLQuery.Select(TABLE_WRITABLEPATH)
                .setWhere(FIELD_WRITABLEPATH_PATH + "=?", path.toString());
    }

    @Override
    public ContentValues getValues()
    {
        ContentValues contentValues = new ContentValues();

        contentValues.put(FIELD_WRITABLEPATH_TITLE, title);
        contentValues.put(FIELD_WRITABLEPATH_PATH, path.toString());

        return contentValues;
    }

    @Override
    public void reconstruct(SQLiteDatabase db, KuickDb kuick, ContentValues item)
    {
        this.title = item.getAsString(FIELD_WRITABLEPATH_TITLE);
        this.path = Uri.parse(item.getAsString(FIELD_WRITABLEPATH_PATH));
    }

    @Override
    public void onCreateObject(SQLiteDatabase db, KuickDb kuick, Object parent, Progress.Listener listener)
    {

    }

    @Override
    public void onUpdateObject(SQLiteDatabase db, KuickDb kuick, Object parent, Progress.Listener listener)
    {

    }

    @Override
    public void onRemoveObject(SQLiteDatabase db, KuickDb kuick, Object parent, Progress.Listener listener)
    {

    }
}

File: app/src/main/java/com/genonbeta/TrebleShot/migration/db/dataobject/WritablePathObjectV12.kt
/*
 * Copyright (C) 2020 Veli Tasalı
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package com.genonbeta.TrebleShot.migration.db.dataobject

import android.net.Uri

/**
 * created by: Veli
 * date: 16.02.2018 12:56
 */
class WritablePathObjectV12 : DatabaseObject<Any?> {
    var title: String? = null
    var path: Uri? = null

    constructor() {}
    constructor(path: Uri?) {
        this.path = path
    }

    constructor(title: String?, path: Uri?) : this(path) {
        this.title = title
    }

    override fun getWhere(): SQLQuery.Select {
        return SQLQuery.Select(v12.Companion.TABLE_WRITABLEPATH)
            .setWhere(v12.Companion.FIELD_WRITABLEPATH_PATH + "=?", path.toString())
    }

    override fun getValues(): ContentValues {
        val contentValues = ContentValues()
        contentValues.put(v12.Companion.FIELD_WRITABLEPATH_TITLE, title)
        contentValues.put(v12.Companion.FIELD_WRITABLEPATH_PATH, path.toString())
        return contentValues
    }

    override fun reconstruct(db: SQLiteDatabase, kuick: KuickDb, item: ContentValues) {
        title = item.getAsString(v12.Companion.FIELD_WRITABLEPATH_TITLE)
        path = Uri.parse(item.getAsString(v12.Companion.FIELD_WRITABLEPATH_PATH))
    }

    override fun onCreateObject(db: SQLiteDatabase, kuick: KuickDb, parent: Any, listener: Progress.Listener) {}
    override fun onUpdateObject(db: SQLiteDatabase, kuick: KuickDb, parent: Any, listener: Progress.Listener) {}
    override fun onRemoveObject(db: SQLiteDatabase, kuick: KuickDb, parent: Any, listener: Progress.Listener) {}
}